### ISSUE TITLE
The project doesn't compile if noFallthroughCasesInSwitch flag is set

### ISSUE THREAD

**Issue by ZixZux** [juanjoDiaz/streamparser-json#26]
The project with this package included doesn't compile if 'noFallthroughCasesInSwitch' flag is set to 'true' in tsconfig.json.

**Comment by juanjoDiaz** [comment]
Hi @ZixZux ,

Typescript recommends using [skipLibCheck](https://www.typescriptlang.org/tsconfig#Completeness_6257) to avoid this kind of issue.

I will fix it in the next release in any case.

**Comment by ZixZux** [comment]
Hi,

The flag skipLibCheck doesn't help is this case, because it only seems to work only for d.ts files: "Skip type checking of declaration files. [...] Rather than doing a full check of all d.ts files, TypeScript will type check the code you specifically refer to in your app’s source code."

But the errors are in your source typescript-files.

In common, source typescript-files shouldn't be included in the distributed package.

**Comment by RobRendell** [comment]
I have the same issue, failing to build because of my more strict compilerOptions - `noFallthroughCasesInSwitch` and `noUncheckedIndexedAccess` in my case.

The [suggested workaroud](https://github.com/microsoft/TypeScript/issues/47387) is that "the package owner [should] release JavaScript and declaration files, not TypeScript files."  What's interesting is that you *do* release JS and .d.ts files, and as far as I can see, correctly reference them in your package.json - you just also happen to bundle the src folder as well.  So I don't know why Typescript is even inspecting your package's source when it should just be using the type declaration files.  I added a comment to the issue I linked to above pointing out this apparent inconsistency.

Anyway, if you're going to omit the typescript files in the next release, that would prevent this particular issue.  Looking forward to it!
